test/e2e/authorizer/reverse_permission_claims_authorizer_test.go

```diff
@@ -0,0 +1,253 @@
+					},
+					Verbs: apisv1alpha1.Verbs{
+						Claimed:    []string{"*"},
+						RestrictTo: []string{"list", "get", "watch"},
+					},
+				},
+			},
+		},
+	}
+	_, err = serviceProviderKcpClient.Cluster(serviceProviderWorkspace).ApisV1alpha1().APIExports().Create(ctx, cowboysAPIExport, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("grant user-2 to be able to bind cowboys API export \"today-cowboys\" from workspace %q", serviceProviderWorkspace)
+	cr, crb := createClusterRoleAndBindings(
+		"user-2-bind",
+		"user-2", "User",
+		[]string{"bind"},
+		"apis.kcp.dev", "apiexports", "today-cowboys",
+	)
+	_, err = serviceProviderKubeClient.Cluster(serviceProviderWorkspace).RbacV1().ClusterRoles().Create(ctx, cr, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = serviceProviderKubeClient.Cluster(serviceProviderWorkspace).RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("create an APIBinding in tenant workspace %q that points to the today-cowboys export from %q", tenantWorkspace, serviceProviderWorkspace)
+	apiBinding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cowboys",
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					Path:       serviceProviderWorkspace.String(),
+					ExportName: "today-cowboys",
+				},
+			},
+			PermissionClaims: []apisv1alpha1.AcceptablePermissionClaim{
+				{
+					PermissionClaim: apisv1alpha1.PermissionClaim{
+						GroupResource: apisv1alpha1.GroupResource{Group: "", Resource: "configmaps"},
+						ResourceSelector: []apisv1alpha1.ResourceSelector{
+							{Namespace: "claimed-namespace", Name: "claimed-configmap"},
+						},
+						Verbs: apisv1alpha1.Verbs{
+							Claimed:    []string{"*"},
+							RestrictTo: []string{"list", "get", "watch"},
+						},
+					},
+					State: apisv1alpha1.ClaimAccepted,
+				},
+			},
+		},
+	}
+	framework.Eventually(t, func() (bool, string) {
+		_, err = tenantKcpClient.Cluster(tenantWorkspace).ApisV1alpha1().APIBindings().Create(ctx, apiBinding, metav1.CreateOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("error creating API binding: %v", err)
+		}
+		return true, ""
+	}, wait.ForeverTestTimeout, time.Millisecond*100, "api binding creation failed")
+
+	t.Logf("wait for API binding %q to have applied permission claims in tenant workspace %q", apiBinding.Name, tenantWorkspace)
+	_, err = tenantKubeClient.Cluster(tenantWorkspace).CoreV1().Namespaces().Create(ctx, &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{Name: "claimed-namespace"},
+	}, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("create configmap \"claimed-configmap\" in namespace \"claimed-namespace\" in tenant workspace %q", tenantWorkspace)
+	_, err = tenantKubeClient.Cluster(tenantWorkspace).CoreV1().ConfigMaps("claimed-namespace").Create(ctx, &v1.ConfigMap{
```

Comment:
Isn't the export restricting the tenant to only list, get, and watch? How does this call succeed?

Reply:
That'a a very unfortunate side-effect of kubernetes RBAC which I found 😞 https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources:

> Note: You cannot restrict create or deletecollection requests by their resource name. For create, this limitation is because the name of the new object may not be known at authorization time. If you restrict list or watch by resourceName, clients must include a metadata.name field selector in their list or watch request that matches the specified resourceName in order to be authorized. For example, kubectl get configmaps --field-selector=metadata.name=my-configmap

I really don't have a very good solution here currently 🤔

Reply:
Concretely, the `create` verb reaches the authorizer but the resource name is not available. Potentially we could lock down all creates in the namespace and I can add an e2e test for this + documentation.

Reply:
So when it comes to `creates` and `deletecollection` the closest we can restrict is by `GroupResource`+`namespace`.

Reply:
My (maybe naive) reading of the permission claim here is: "a user in the binding workspace may only list+get+watch a configmap named claimed-configmap in the namespace claimed-namespace". I guess that's not correct?

Reply:
Your assumption is not correct indeed. All other resources are owned by the user. Only the resources matching the resource selector must be restricted.

Reply:
So using your term the logic is rather "a user in the binding workspace may only list+get+watch a configmap named claimed-configmap in the namespace claimed-namespace but the user can do anything on all other resources."

Reply:
Gotcha, thanks